rcsb/utils/chemref/RcsbLigandReferenceGenerator.py

```diff
@@ -0,0 +1,300 @@
+##
+#  File:           RcsbLigandReferenceGenerator.py
+#  Date:           2025-12-10 Chenghua Shao
+#
+#  Update:
+##
+"""
+Generate ligand quality reference data that is used for ligand quality score computation.
+
+"""
+import csv
+import logging
+import numpy as np
+from sklearn.decomposition import PCA
+from sklearn.preprocessing import StandardScaler
+from rcsbapi.data import DataQuery as Query
+from rcsbapi.data import ALL_STRUCTURES
+from rcsbapi.config import config
+from rcsb.utils.io.StashableBase import StashableBase
+
+logger = logging.getLogger(__name__)
+
+
+class RcsbLigandReferenceGenerator(StashableBase):
+    """ This class generates ligand quality reference data by performing the following steps:
+    Query ligand quality metrics from RCSB GraphQL API;
+    Process the ligand quality data by filtering, aggregating, and formatting;
+    Generate ligand reference data through PCA;
+        Attributes:
+            data: Stateful data updated with each step
+        Methods:
+            query(pdb_id): Fetch ligand quality metrics for given PDB IDs.
+            filter(): Filter the queried ligand quality scores based on defined criteria.
+            reduce(): Reduce the data by combining multiple instances of the same ligand in the same PDB entry.
+            analyze(): Run PCA on the filtered and reduced ligand quality scores to generate reference data.
+            generate(pdb_ids): Full pipeline to generate ligand quality reference data by running the steps of
+                query -> filter -> reduce -> analyze.
+            writeReference(output_file): Write the generated ligand quality reference data to a csv file.
+    """
+    def __init__(self):
+        self.data = None
+        # Increase some settings to speed up querying
+        config.DATA_API_MAX_CONCURRENT_REQUESTS = 15
+        config.DATA_API_REQUESTS_PER_SECOND = 30
+
+    def generate(self, pdb_ids: list[str] = []) -> "RcsbLigandReferenceGenerator":
```

Comment:
For this line and line 80, the linter is showing [this warning](https://pylint.pycqa.org/en/latest/user_guide/messages/warning/dangerous-default-value.html). Can you replace the `= []` with `= None`, and then reassign to an empty list in the method? 

The simple approach would be:

```python
    def generate(self, pdb_ids: list[str] = None) -> "RcsbLigandReferenceGenerator":
        pdb_ids = [] if pdb_ids is None else pdb_ids
```